src/Adapter/MSTestAdapter.PlatformServices/Services/ReflectionOperations2.cs

```diff
@@ -23,26 +26,26 @@ public ReflectionOperations2()
 #pragma warning disable IL2026 // Members attributed with RequiresUnreferencedCode may break when trimming
 #pragma warning disable IL2067 // 'target parameter' argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to 'target method'.
 #pragma warning disable IL2057 // Unrecognized value passed to the typeName parameter of 'System.Type.GetType(String)'
-    public IEnumerable<ConstructorInfo> GetDeclaredConstructors(Type classType)
-        => classType.GetTypeInfo().DeclaredConstructors;
+    public ConstructorInfo[] GetDeclaredConstructors(Type classType)
+        => classType.GetConstructors(DeclaredOnlyLookup);
```

Comment:
`GetTypeInfo` has an extra unnecessary allocation, and forces us to use IEnumerable which will cause an enumerator allocation at callsites as well.